src/NHibernate/SqlCommand/SqlInsertBuilder.cs

```diff
@@ -221,7 +220,7 @@ public SqlCommandInfo ToSqlCommandInfo()
 
 		public SqlType[] GetParametersTypeArray()
 		{
-			return (new List<SqlType>(new SafetyEnumerable<SqlType>(columns.Values))).ToArray();
```

Comment:
There is a subtle difference between `SafetyEnumerable` and `OfType`. `SafetyEnumerable` will not filter out nulls, while `OfType` will. Not sure how this is relevant here.

Reply:
Good catch.. Better to restore original logic